src/server/editorServices.ts

```diff
@@ -17,6 +17,58 @@ namespace ts.server {
         (event: ProjectServiceEvent): void;
     }
 
+    const indentStyle = createMap({
+        "none": IndentStyle.None,
+        "block": IndentStyle.Block,
+        "smart": IndentStyle.Smart
+    });
+
+    export function convertFormatOptions(protocolOptions: protocol.FormatCodeSettings): FormatCodeSettings {
+        if (typeof protocolOptions.indentStyle === "string") {
+            protocolOptions.indentStyle = indentStyle[protocolOptions.indentStyle.toLowerCase()];
+            Debug.assert(protocolOptions.indentStyle !== undefined);
+        }
+        return <any>protocolOptions;
+    }
+
+    export function convertCompilerOptions(protocolOptions: protocol.ExternalProjectCompilerOptions): CompilerOptions & protocol.CompileOnSaveMixin {
+        protocolOptions.target = convertCompilerOptionEnum(targetCommandLineOption, protocolOptions.target);

```

Comment:
now if i add a new compiler option, i need to add it to commandline-parser and here as well. can i do it in one place?